src/events/event_source.cpp

```diff
@@ -0,0 +1,14 @@
+//
+// Created by doralitze on 11/15/23.
+//
+
+#include "event_source.hpp"
+#include "event_storage.hpp"
+
+namespace dmxfish::events {
+    event_source::event_source() : sender_id{} {}
+
+    void event_source::deregister() {
+
+    }
```

Comment:
is it right that this function does nothing?

Reply:
Yes, it hasn't been implemented yet. I forgot to do so and the unit test apparently lacks a proper check for this.